kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/DefaultKroxyliciousTester.java

```diff
@@ -60,12 +81,30 @@ private KroxyliciousClients clients() {
     }
 
     private KroxyliciousClients clients(String virtualCluster) {
-        return clients.computeIfAbsent(virtualCluster, this::buildKroxyliciousClients);
+        return clients.computeIfAbsent(virtualCluster,
+                k -> clientFactory.build(virtualCluster, buildDefaultClientConfiguration(virtualCluster)));
     }
 
     @NonNull
-    private KroxyliciousClients buildKroxyliciousClients(String clusterName) {
-        return this.clientFactory.build(clusterName, KroxyliciousConfigUtils.bootstrapServersFor(clusterName, kroxyliciousConfig));
+    private Map<String, Object> buildDefaultClientConfiguration(String virtualCluster) {
+        Map<String, Object> defaultClientConfig = new HashMap<>();
+        defaultClientConfig.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, KroxyliciousConfigUtils.bootstrapServersFor(virtualCluster, kroxyliciousConfig));
+        configureClientTls(virtualCluster, defaultClientConfig);
+        return defaultClientConfig;
+    }
+
+    private void configureClientTls(String virtualCluster, Map<String, Object> defaultClientConfig) {
+        final VirtualCluster definedCluster = kroxyliciousConfig.virtualClusters().get(virtualCluster);
+        if (definedCluster != null) {
+            final Optional<Tls> tls = definedCluster.tls();
+            if (tls.isPresent() && trustStoreConfiguration.isPresent()) {
+                final KroxyliciousTesterBuilder.TrustStoreConfiguration storeConfiguration = trustStoreConfiguration.get();
+                defaultClientConfig.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SSL.name);
+                defaultClientConfig.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, storeConfiguration.trustStoreLocation());
+                defaultClientConfig.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, storeConfiguration.trustStorePassword());
+            }
```

Comment:
That's only true if self signed certificates are in use.   I know that all we cater for at the moment, but I'd still write the code to support otherwise.  
That's only true if self signed certificates are in use.   I know that all we cater for at the moment, but I'd still write the code to support otherwise.  
You are assuming self signed... I know that's all we support now, but might as well not hardcode a bad assumption.

```suggestion
            if (tls.isPresent()) {
                defaultClientConfig.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,                 final 
                 if (tls.isPresent() && trustStoreConfiguration.isPresent()) {

                    KroxyliciousTesterBuilder.TrustStoreConfiguration storeConfiguration = trustStoreConfiguration.get();
SecurityProtocol.SSL.name);
                   defaultClientConfig.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, storeConfiguration.trustStoreLocation());
                   defaultClientConfig.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, storeConfiguration.trustStorePassword());
                }
            }
```